src/main/java/jsr223/docker/file/DockerFileScriptEngine.java

```diff
@@ -394,11 +497,15 @@ private void handleShutdown(ScriptContext context) {
             processRun.destroy();
         }
 
-        if (containerStarted) {
+        if (processExec != null) {
+            processExec.destroy();
+        }
+
+        if (containerStarted && dockerActions.contains("stop")) {
             stopAndRemoveContainer(containerTagName, context);
         }
 
-        if (imageCreated && !DockerFilePropertyLoader.getInstance().isKeepDockerFile()) {
+        if (imageCreated && dockerActions.contains("rmi")) {
```

Comment:
That would be nice to create final variables for "rmi" and "stop".

Reply:
okay